Makefile

```diff
@@ -20,36 +21,53 @@ $(info !!!!! Defaulting to Linux x86_64 build options !!!!!)
 BUILD_OPT_EXE = $(BUILD_OPT_EXE.Linux.x86_64)
 endif
 
-pnut-sh: pnut.c sh.c sh-runtime.c
-	mkdir -p $(BUILD_DIR)
-	gcc $(BUILD_OPT_SH) pnut.c -o $(BUILD_DIR)/pnut-sh
+.PHONY: default install uninstall clean test-sh test-i386-linux test-x86_64-linux test-x86_64-mac
+
+default: $(BUILD_DIR)/pnut-sh $(BUILD_DIR)/pnut-exe
+
+all: $(BUILD_DIR)/pnut-sh \
+	$(BUILD_DIR)/pnut-sh.sh \
+	$(BUILD_DIR)/pnut-sh-bootstrapped.sh \
+	$(BUILD_DIR)/pnut-exe \
+	$(BUILD_DIR)/pnut-exe.sh \
+	$(BUILD_DIR)/pnut-exe-bootstrapped
+
+$(BUILD_DIR)/config.mk:
+	./configure
+
+include $(BUILD_DIR)/config.mk
```

Comment:
`./configure` scripts are usually called before using `make` but it looks like the Makefile creates it if it's missing (but only when `make install` is used?) so I'm not sure what's the expected usage?

Also, when calling `make` without first calling the `configure` script, it complains with `Makefile:38: build/config.mk: No such file or directory`.